### PR TITLE
2.x fix publishing pluging

### PR DESCRIPTION
**Pull request by KovalevAndrey**
## Description

Fixes gradle task to work correctly with multiptaform.

1. Previously it was creating one publication for kotlin multiplatfrom module instead of one per target
2. Refactored so most of the code is reused across Android, JavaLibrary and KMP publications
3. Created extension so we can explicitly provide artifactId for publications

Local maven publishing result:

Screenshot 2023-04-27 at 09 25 46

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: appyx-components/demos/common/build.gradle.kts [hunk 1]
```diff
@@ -6,6 +6,10 @@ plugins {
     id("appyx-publish-multiplatform")
 }
 
+publishingPlugin {
```
Comment (CherryPerry) [inline appyx-components/demos/common/build.gradle.kts]: Why publish demos?

File: plugins/publish-plugin/src/main/kotlin/AndroidAppyxPublishPlugin.kt [hunk 2]
```diff
@@ -14,5 +18,16 @@ internal class AndroidAppyxPublishPlugin : ProjectPlugin() {
         }
     }
 
+    override fun PublicationContainer.createPublications(project: Project) {
+        create<MavenPublication>("appyxRelease") {
```
Comment (CherryPerry) [inline plugins/publish-plugin/src/main/kotlin/AndroidAppyxPublishPlugin.kt]: Let's sync with MPP naming – `androidRelease`

File: plugins/publish-plugin/src/main/kotlin/JavaAppyxPublishPlugin.kt [hunk 3]
```diff
@@ -10,5 +14,16 @@ internal class JavaAppyxPublishPlugin : ProjectPlugin() {
         }
     }
 
+    override fun PublicationContainer.createPublications(project: Project) {
+        create<MavenPublication>("appyxRelease") {
```
Comment (CherryPerry) [inline plugins/publish-plugin/src/main/kotlin/JavaAppyxPublishPlugin.kt]: Let's sync with MPP naming – `jvm`

File: plugins/publish-plugin/src/main/kotlin/MultiplatformAppyxPublishPlugin.kt [hunk 4]
```diff
@@ -1,12 +1,37 @@
...
+        val javadocJar by project.tasks.registering(Jar::class) {
+            archiveClassifier.set("javadoc")
+        }
+
+        val artifactId = project.extensions.getByType(ProjectPluginExtension::class.java).artifactId
+
+        mppExtension.targets.all {
+            mavenPublication {
+                if (artifactId.isNotEmpty()) {
+                    this.artifactId = "$artifactId-$targetName"
```
Comment (CherryPerry) [inline plugins/publish-plugin/src/main/kotlin/MultiplatformAppyxPublishPlugin.kt]: I hope it works as expected.

**Review by CherryPerry — APPROVED** [review]

File: plugins/publish-plugin/src/main/kotlin/MultiplatformAppyxPublishPlugin.kt (hunk 4 above)
Comment (KovalevAndrey) [reply plugins/publish-plugin/src/main/kotlin/MultiplatformAppyxPublishPlugin.kt]: it does - check the artifacts on the image in the description

File: appyx-components/demos/common/build.gradle.kts (hunk 1 above)
Comment (KovalevAndrey) [reply appyx-components/demos/common/build.gradle.kts]: we want to showcase some stuff but we won't commit to supporting it

**Review by CherryPerry — APPROVED** [review]

**Pull request merged**